packages/redux-simple-auth/src/middleware.js

```diff
@@ -120,6 +126,35 @@ export default (config = {}) => {
             return response
           })
         }
+        case INVALIDATE_SESSION: {
+          const state = getState()
+          const authenticatorName = getAuthenticator(state)
+
+          if (authenticatorName) {
+            const authenticator = findAuthenticator(authenticatorName)
+
+            if (!authenticator) {
+              throw new Error(
+                `No authenticator with name \`${action.meta.authenticator}\` ` +
+                  'was found. Be sure you have defined it in the authenticators ' +
+                  'config.'
+              )
+            }
+
+            if (authenticator) {
+              authenticator.invalidate(getSessionData(state)).then(
+                // dispatch session invalidation to allow things outside this
+                // direct ecosystem to be updated.
+                () => next(action),
+                () => dispatch(invalidateSessionFailed())
+              )
+            } else {
```

Comment:
This `else` case will never be hit since an error will be `throw`n before getting here.

That does beg the question, does it make sense to `throw` or return a rejected promise with the dispatched  `invalidateSessionFailed()` action?

In this case, I think throwing probably makes more sense since its likely a development error if you're trying to invalidate a session on an authenticator that doesn't exist. With that, this `else` can be removed.

Reply:
Hmm, so just a plain throw?